client/apps/admin/components/application_instance_settings/xml_config.jsx

```diff
@@ -1,35 +1,28 @@
 import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
 import _ from 'lodash';
 import * as ApplicationInstanceActions from '../../actions/application_instances';
 import Textarea from '../common/textarea';
+import { useDispatch, useSelector } from 'react-redux';
 
-const select = (state, props) => (
-  {
-    loading: state.applicationInstances.loading,
-    loaded: state.applicationInstances.loaded,
-    applicationInstances: _.filter(
-      state.applicationInstances.applicationInstances,
-      { application_id: parseInt(props.params.applicationId, 10) }
-    ),
-  }
-);
-
-export function XmlConfig(props) {
+export default function XmlConfig(props) {
   const {
-    loading,
-    loaded,
     params,
-    applicationInstances
   } = props;
 
+  const dispatch = useDispatch();
+  const loading = useSelector((state) => state.applicationInstances.loading);
+  const loaded = useSelector((state) => state.applicationInstances.loaded);
+  const applicationInstances = useSelector((state) => state.applicationInstances);
```

Comment:
This should only include the application instances for the matching application.

For example:
applicationInstances: _.filter(
      state.applicationInstances.applicationInstances,
      { application_id: parseInt(props.params.applicationId, 10) }
    ),

Reply:
I added the filter.